### PR TITLE
[IMP] mail: allow guests to switch between channels

### PR DESCRIPTION
**Pull request by tsm-odoo**
Guests can be invited to channel with an invitation link that
leads to the discuss standalone page. Until now, guests could
only see one channel: the one of the invitation link.

However, guests can be linked to multiple discuss channels
through multiple invitation links, live chats... Currently,
there is no easy way for a guest to navigate accross those
channels.

This PR enable the discuss side bar in the standalone page.

task-4016335

### CODE REVIEW

File: addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.DiscussSidebarCategory" t-inherit-mode="extension">
+        <xpath expr="//*[@t-ref='categoryHeader']" position="after">
+            <div t-if="state.editing === category.id" class="p-2" t-ref="selector">
+                <ChannelSelector category="category" onValidate.bind="stopEditing" autofocus="true" close.bind="stopEditing" />
+            </div>
+        </xpath>
+        <xpath expr="//*[@t-ref='openCategoryBtn']" position="after">
```
Comment (alexkuhn) [inline addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml:9]: Nitpicking: prefer `name` when only used in xpath

File: addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.DiscussSidebarCategory" t-inherit-mode="extension">
+        <xpath expr="//*[@t-ref='categoryHeader']" position="after">
```
Comment (alexkuhn) [inline addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml:4]: same

File: addons/mail/static/src/discuss/core/public_web/discuss_sidebar_categories.xml [hunk 3]
```diff
@@ -11,30 +11,17 @@
     </t>
 
     <t t-name="mail.DiscussSidebarCategory">
-        <div class="o-mail-DiscussSidebarCategory d-flex align-items-center my-1" t-att-class="category.extraClass">
-            <button class="btn btn-link text-reset flex-grow-1 d-flex align-items-baseline mx-1 p-0 text-start opacity-100-hover opacity-75" t-on-click="() => this.toggleCategory(category)">
+        <div class="o-mail-DiscussSidebarCategory d-flex align-items-center my-1" t-att-class="category.extraClass" t-ref="categoryHeader">
+            <button class="btn btn-link text-reset flex-grow-1 d-flex align-items-baseline mx-1 p-0 text-start opacity-100-hover opacity-75" t-on-click="() => this.toggleCategory(category)" t-ref="openCategoryBtn">
```
Comment (alexkuhn) [inline addons/mail/static/src/discuss/core/public_web/discuss_sidebar_categories.xml:15]: The template is named `DiscussSidebarCategory`, using `t-ref`/`name` that has "category" in the name is redundant.

I would rename `categoryHeader` and `openCategoryBtn` to respectively `header` and `toggler`.

File: addons/mail/static/tests/tours/discuss_channel_public_tour.js [hunk 4]
```diff
@@ -5,7 +5,8 @@ registry.category("web_tour.tours").add("discuss_channel_public_tour.js", {
     test: true,
     steps: () => [
         {
-            trigger: ".o-mail-DiscussPublic",
+            trigger: ".o-mail-Discuss",
+            timeout: 10000000000,
```
Comment (alexkuhn) [inline addons/mail/static/tests/tours/discuss_channel_public_tour.js:9]: I assume the `timeout` is for debug? If so, remove

File: addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml:9]: If you really don't want ref, the 2nd best way is to use data- attribute.

Comment (tsm-odoo) [reply addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml:9]: I don't like polluting the DOM for XPath purposes either.

Comment (alexkuhn) [reply addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml:9]: No one really care for `name` polluting the DOM. `t-ref` is fine but this introduces some OWL internal processing, which could causes subtle issues.

`name` attribute is much simpler and to the point, hence why I'd prefer it. But if you insist with `t-ref` then fine, both should work the same so it doesn't really matter.

Comment (tsm-odoo) [reply addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.xml]: Indeed, not a big deal. I applied your suggestion.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**